### PR TITLE
feat(rulesets): add support for PR labeled events

### PR DESCRIPTION
**Pull request by wsan3**
## Description
Request for Vela to process and create builds for pull request [labeled](https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=labeled#pull_request) and [unlabeled](https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=unlabeled#pull_request) webhooks.

## Related
GitHub [issue #913](https://github.com/go-vela/community/issues/913)
GitHub [ideas #768](https://git.target.com/orgs/vela/projects/36/views/1?filterQuery=win+san&pane=issue&itemId=19205)

### CODE REVIEW

File: src/elm/Pages/RepoSettings.elm [hunk 1]
```diff
@@ -347,6 +347,16 @@ events repo msg =
                         allowEvents.pull.reopened
                       <|
                         msg repo.org repo.name repo "allow_pull_reopened"
+                    , checkbox "Labled"
+                        "allow_pull_labeled"
+                        allowEvents.pull.labeled
+                      <|
+                        msg repo.org repo.name repo "allow_pull_labeled"
+                    , checkbox "Unlabled"
```
Comment (plyr4) [inline src/elm/Pages/RepoSettings.elm:355]:
```suggestion
                    , checkbox "Unlabeled"
```

File: src/elm/Pages/RepoSettings.elm [hunk 2]
```diff
@@ -347,6 +347,16 @@ events repo msg =
                         allowEvents.pull.reopened
                       <|
                         msg repo.org repo.name repo "allow_pull_reopened"
+                    , checkbox "Labled"
```
Comment (plyr4) [inline src/elm/Pages/RepoSettings.elm:350]:
```suggestion
                    , checkbox "Labeled"
```

**Review by plyr4 — CHANGES_REQUESTED** [review]
nice!

File: elm.json [hunk 3]
```diff
@@ -25,7 +25,7 @@
       "jzxhuang/http-extras": "2.1.0",
       "krisajenkins/remotedata": "6.0.1",
       "pablen/toasty": "1.2.0",
-      "ryannhg/date-format": "2.3.0",
+      "ryan-haskell/date-format": "1.0.0",
```
Comment (plyr4) [inline elm.json:28]: nice, it got renamed, but i think this should still be `2.3.0` https://github.com/ryan-haskell/date-format/releases

**Review by plyr4 — CHANGES_REQUESTED** [review]

File: elm.json (hunk 3 above)
Comment (wsan3) [reply elm.json:28]: The repo owner suggested using `v1.0.0` in this thread https://discourse.elm-lang.org/t/ryannhg-packages-renamed-to-ryan-haskell/9705

File: src/elm/Pages/RepoSettings.elm (hunk 2 above)
Comment (wsan3) [reply src/elm/Pages/RepoSettings.elm:350]: Good catch

File: src/elm/Pages/RepoSettings.elm (hunk 1 above)
Comment (wsan3) [reply src/elm/Pages/RepoSettings.elm:355]: Good catch

File: elm.json (hunk 3 above)
Comment (plyr4) [reply elm.json:28]: excellent! ignore me!

**Review by plyr4 — APPROVED** [review]

**Review by KellyMerrick — APPROVED** [review]
lgtm

**Pull request merged**